File: code/Python/Chapter1/1-3/monte_pi.py
import random
import math

def monte_pi(n):
    cnt = 0
    for _ in range(n):
        x = random.random()     # [0, 1)
        y = random.random()     # [0, 1)
        if math.sqrt(x**2 + y**2) <= 1:
            cnt += 1
    return 4.0 * cnt / n

print(monte_pi(10000000))

File: code/Java/Chapter1/1-3/MontePI.java
public class MontePI {
	public static void main(String[] args) {
		System.out.println("PI = " + montePI(10000000));
	}
	
	public static double montePI(int n) {
		int cnt = 0;
		for(int i = 0; i < n; i++) {
	        double x = Math.random();		// [0, 1)
	        double y = Math.random();		// [0, 1)
	        if(Math.sqrt(x*x + y*y) <= 1) {
	            cnt++;
	        }
	    }
	    return 4.0 * cnt / n;
	}
}